doc/developers/contributing.rst

```diff
@@ -676,20 +676,226 @@ We are glad to accept any sort of documentation:
+    * When specifying the dtype of an ndarray, use e.g. ``dtype=np.int32`` after
+      defining the shape: ``ndarray of shape (n_samples,), dtype=np.int32``. You
+      can specify multiple dtype as a set: ``array-like of shape (n_samples,),
+      dtype={np.float64, np.float32}``. If one wants to mention arbitrary
+      precision, use `integral` and `floating` rather than the Python dtype
+      `int` and `float`. When both `int` and `floating` are supported, there is
+      no need to specify the dtype.
+    * When the default is ``None``, ``None`` only needs to be specified at the
+      end with ``default=None``. Be sure to include in the docstring, what it
+      means for the parameter or attribute to be ``None``.
+
+* Add "See Also" in docstrings for related classes/functions.
+
+* "See Also" in docstrings should be one line per reference, with a colon and an
+  explanation, for example::
+
+    See Also
+    --------
+    SelectKBest : Select features based on the k highest scores.
+    SelectFpr : Select features based on a false positive rate test.
+
+* Add one or two snippets of code in "Example" section to show how it can be used.
+
+|details-end|
+
+|details-start|
+**Guidelines for writing the user guide and other reStructuredText documents**
+|details-split|
+
+It is important to keep a good compromise between mathematical and algorithmic
+details, and give intuition to the reader on what the algorithm does.
+
+* Begin with a concise, hand-waving explanation of what the algorithm/code does on
+  the data.
+
+* Highlight the usefulness of the feature and its recommended application.
+  Consider including the algorithm's complexity
+  (:math:`O\left(g\left(n\right)\right)`) if available, as machine-dependent
+  "rules of thumb" can be very machine-dependent. If those complexities are not
+  available, then rules of thumb may be provided instead.
+
+* Incorporate a relevant figure (generated from an example) to provide intuitions.
+
+* Include one or two short code examples to demonstrate the feature's usage.
+
+* Introduce any necessary mathematical equations, followed by references. By
+  deferring the mathematical aspects, the documentation becomes more accessible
+  to users primarily interested in understanding the feature's practical
+  implications rather than its underlying mechanics.
+
+* When editing reStructuredText (``.rst``) files, try to keep line length under
+  88 characters when possible (exceptions include links and tables).
+
+* In scikit-learn reStructuredText files both single and double backticks
+  surrounding text will render as inline literal (often used for code, e.g.,
+  `list`). This is due to specific configurations we have set. Single
+  backticks should be used nowadays.
+
+* Too much information makes it difficult for users to access to the
+  content they are interested in. Use dropdowns to hide by default:
+
+  - low hierarchy sections such as `Examples`, `References`, `Properties`, etc.
```

Comment:
After a discussion with @ogrisel that was inline with @GaelVaroquaux, let's not fold Examples. Instead, we should advocate for making sure that the Examples section comes right after the main discussion with the least possible folded section in-between.

Reply:
Added a small mention on this regard.